docs/manuals/gui/viselements/charts/heatmap.md

```diff
@@ -0,0 +1,223 @@
+## Heatmap charts
+
+A heatmap depicts values for a main variable of interest across two axis variables as a grid of colored squares. The axis variables are divided into ranges like a bar chart or histogram, and each cell’s color indicates the value of the main variable in the corresponding cell range.
+
+### When to Use?
+- To show user behavior on specific webpages.
+- To display the magnitude of a data set over two dimensions.
+- In retail matrix, manufacturing diagram, and population maps.
+- For marketing goals and analytics, reflecting on user behavior on specific webpages.
+- And more
+
+### Basic Heatmap
+The chart below shows the colors in accordance with each temperatures
+
+```py
+from taipy.gui import Gui
+
+data = {
+    "temperature": [[1, 20, 30], [20, 1, 60], [30, 60, 1]],
+}
+options = {'colorscale': 'Reds'}
+md = """
+## Basic Heatmap
+<|{data}|chart|z=temperature|type=heatmap|options={options}|>
+"""
+
+Gui(md).run()
+```
+![Basic Heatmap](heatmap_basic.png)
+
+### Heatmap with Categorical Axis Labels
+In the following example, we have an array which defines the data (harvest by different farmers in tons/year) to color code. We then also need two lists of names of farmers and vegetables cultivated by them.
+```py
+from taipy.gui import Gui
+import pandas as pd
+
+data = [
+    pd.DataFrame({
+        'harvest': [
+            [0.8, 2.4, 2.5, 3.9, 0.0, 4.0, 0.0],
+            [2.4, 0.0, 4.0, 1.0, 2.7, 0.0, 0.0],
+            [1.1, 2.4, 0.8, 4.3, 1.9, 4.4, 0.0],
+            [0.6, 0.0, 0.3, 0.0, 3.1, 0.0, 0.0],
+            [0.7, 1.7, 0.6, 2.6, 2.2, 6.2, 0.0],
+            [1.3, 1.2, 0.0, 0.0, 0.0, 3.2, 5.1],
+            [0.1, 2.0, 0.0, 1.4, 0.0, 1.9, 6.3]
+        ],
+        'vegetables': ["cucumber",
+                       "tomato",
+                       "lettuce",
+                       "asparagus",
+                       "potato",
+                       "wheat",
+                       "barley"]
+    }),
+    pd.DataFrame({
+        'farmers': ["Farmer Joe",
+                    "Upland Bros.",
+                    "Smith Gardening",
+                    "Agrifun",
+                    "Organiculture",
+                    "BioGoods Ltd.",
+                    "Cornylee Corp."],
+    })
+]
+options = {'colorscale': 'Rainbow'}
+md = """
+## Heatmap with Categorical Axis Labels
+<|{data}|chart|type=heatmap|z=0/harvest|x=1/farmers|y=0/vegetables|options={options}|>
+"""
+
+Gui(md).run()
+```
+![Heatmap with Categorical Axis Labels](heatmap_axis_labels.png)
+
+### Annotated Heatmap
+In this chart, we have average temperature (°F) in seasons of top 4 US cities
```

Comment:
Intro + specific:
"You may want to display the actual value that is represented, at the appropriate location in the heatmap.

This example creates a heatmap that represents the temperature (in Fahrenheit) for different locations, in different seasons."